lib/Db/AppointmentConfig.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2021 Anna Larch <anna.larch@gmx.net>
+ *
+ * @author Anna Larch <anna.larch@gmx.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+// @TODO rename and adjust types from migration
+
+namespace OCA\Calendar\Db;
+
+use JsonSerializable;
+use OCP\AppFramework\Db\Entity;
+use function json_decode;
+
+/**
+ * @method int getId()
+ * @method void setId(int $id)
+ * @method string getToken()
+ * @method void setToken(string $token)
+ * @method string getName()
+ * @method void setName(string $name)
+ * @method string getDescription()
+ * @method void setDescription(string $name)
+ * @method string getLocation()
+ * @method void setLocation(?string $name)
+ * @method string getVisibility()
+ * @method void setVisibility(string $visibility)
+ * @method string getUserId()
+ * @method void setUserId(string $userId)
+ * @method string getTargetCalendarUri()
+ * @method void setTargetCalendarUri(string $calendarUri)
+ * @method string|null getCalendarFreebusyUris()
+ * @method void setCalendarFreebusyUris(?string $freebusyUris)
+ * @method string getAvailability()
+ * @method void setAvailability(?string $availability)
+ * @method int getLength()
+ * @method void setLength(int $length)
+ * @method int getIncrement()
+ * @method void setIncrement(int $increment)
+ * @method int getPreparationDuration()
+ * @method void setPreparationDuration(int $prepDuration)
+ * @method int getFollowupDuration()
+ * @method void setFollowupDuration(int $followup)
+ * @method int getBuffer()
+ * @method void setBuffer(int $buffer)
+ * @method int|null getDailyMax()
+ * @method void setDailyMax(?int $max)
+ */
+class AppointmentConfig extends Entity implements JsonSerializable {
+
+	/** @var string */
+	protected $token;
```

Comment:
Is this field used for anything? DB complains about missing default value when creating a new appointment via the Controller -> Service -> Mapper.

Reply:
It's used for the URL and later on for the `X-Nextcloud-Appointment` head in the ICS

Reply:
Then we either need to force a default value via a migration or add it in the PHP code.

Reply:
Is fixed on the upstream branch.